src/tabnote.ts

```diff
@@ -228,21 +240,21 @@ export class TabNote extends StemmableNote {
   }
 
   // Set the `stave` to the note
-  setStave(stave) {
+  setStave(stave: Stave): this {
     super.setStave(stave);
-    this.context = stave.context;
+    this.setContext(stave.getContext());
 
     // Calculate the fret number width based on font used
     let i;
```

Comment:
I think you can remove this, and move the `let` to the beginning of the loop below.

Reply:
done